R/radqlqc.R

```diff
@@ -0,0 +1,1304 @@
+  checkmate::assert_number(number, lower = 1)
+
+  # ADQLQC data -------------------------------------------------------------
+  # prepare ADaM adqlqc data
+  adqlqc1 <- prep_adqlqc(df = QS)
+
+  # derive AVAL and AVALC
+  adqlqc1 <- mutate(
+    adqlqc1,
+    AVAL = as.numeric(.data$QSSTRESC),
+    AVALC = case_when(
+      .data$QSTESTCD == "QSALL" ~ .data$QSREASND,
+      TRUE ~ .data$QSORRES
+    ),
+    AVISIT = .data$VISIT,
+    AVISITN = .data$VISITNUM,
+    ADTM = .data$QSDTC
+  )
+
+  # include scale calculation
+  ADQLQCtmp <- calc_scales(adqlqc1)
+
+  # order to prepare for change from screening and baseline values
+  ADQLQCtmp <- ADQLQCtmp[order(ADQLQCtmp$STUDYID, ADQLQCtmp$USUBJID, ADQLQCtmp$PARAMCD, ADQLQCtmp$AVISITN), ] # nolint
+
+  ADQLQCtmp <- Reduce( # nolint
+    rbind,
+    lapply(
+      split(ADQLQCtmp, ADQLQCtmp$USUBJID),
+      function(x) {
+        x$STUDYID <- ADSL$STUDYID[which(ADSL$USUBJID == x$USUBJID[1])] # nolint
+        x$ABLFL2 <- ifelse(x$AVISIT == "SCREENING", "Y", "") # nolint
+        x$ABLFL <- ifelse( # nolint
+          x$AVISIT == "BASELINE" &
+            x$PARAMCD != "EX028",
+          "Y",
+          ifelse(
+            x$AVISIT == "CYCLE 1 DAY 1" &
+              x$PARAMCD != "EX028",
+            "Y",
+            ""
+          )
+        )
+        x
+      }
+    )
+  )
+
+  ADQLQCtmp$BASE2 <- ifelse(
+    str_detect(ADQLQCtmp$PARCAT2, "Completion", negate = TRUE),
+    random.cdisc.data:::retain(
+      df = ADQLQCtmp,
+      value_var = ADQLQCtmp$AVAL,
+      event = ADQLQCtmp$ABLFL2 == "Y"
+    ),
+    NA
+  ) # nolint
+
+  ADQLQCtmp$BASE <- ifelse(
+    ADQLQCtmp$ABLFL2 != "Y" &
+      str_detect(ADQLQCtmp$PARCAT2, "Completion", negate = TRUE),
+    random.cdisc.data:::retain(
+      ADQLQCtmp,
+      ADQLQCtmp$AVAL,
+      ADQLQCtmp$ABLFL == "Y"
+    ),
+    NA
+  ) # nolint
+
+  ADQLQCtmp <- ADQLQCtmp %>% # nolint
+    dplyr::mutate(CHG2 = .data$AVAL - .data$BASE2) %>%
+    dplyr::mutate(PCHG2 = 100 * (.data$CHG2 / .data$BASE2)) %>%
+    dplyr::mutate(CHG = .data$AVAL - .data$BASE) %>%
+    dplyr::mutate(PCHG = 100 * (.data$CHG / .data$BASE)) %>%
+    random.cdisc.data:::var_relabel(
+      STUDYID = attr(ADSL$STUDYID, "label"),
+      USUBJID = attr(ADSL$USUBJID, "label")
+    )
+
+  # derive CHGCAT1 ----------------------------------------------------------
+  ADQLQCtmp <- derv_chgcat1(dataset = ADQLQCtmp)
+
+
+  ADQLQCtmp <- random.cdisc.data:::var_relabel( # nolint
+    ADQLQCtmp,
+    STUDYID = "Study Identifier",
+    USUBJID = "Unique Subject Identifier"
+  )
+
+  ADQLQCtmp <- arrange(
+    ADQLQCtmp,
+    .data$USUBJID,
+    .data$AVISITN
+  )
+
+  # Merge ADSL --------------------------------------------------------------
+  # adsl variables from ADQLQC in gdsr
+  adsl_vars <- filter(
+    adamv_variables(),
+    .data$dataset == "ADQLQC",
+    str_detect(.data$derivation_text, "^ADSL")
+  )
+  adsl_vars <- adsl_vars$name
```

Comment:
Can we just hardcode this list as 

 [1] "STUDYID"  "USUBJID"  "SUBJID"   "SITEID"   "REGION1"  "COUNTRY"  "ETHNIC"   "AGE"     
 [9] "AGEU"     "AAGE"     "AAGEU"    "AGEGR1"   "AGEGR2"   "AGEGR3"   "STRATwNM" "STRATw"  
[17] "STRATwV"  "SEX"      "RACE"     "ITTFL"    "SAFFL"    "PPROTFL"  "TRT01P"   "TRTxxP"  
[25] "TRT01A"   "TRTxxA"   "TRTSEQP"  "TRTSEQA"  "TRTSDTM"  "TRTSDT"   "TRTEDTM"  "TRTEDT"  
[33] "DCUTDT"

Reply:
Thank you, variables now hardcoded.